### PR TITLE
Feature/issue 74 update widget driver documentation to match new annotations

### PR DESCRIPTION
**Pull request by robert-northmind**
## Description
Solves issue #74 

- Updates all documentation in widget_driver to match new annotation names.
- Updates example app to show case how to use new future-annotation
- Updates the make script for generating new example app code, now it also formats the code according to our formatting rules.

## Type of Change

- [ ] 🚀 New feature (non-breaking change)
- [ ] 🛠️ Bug fix (non-breaking change)
- [ ] ⚠️ Breaking change (feature or bug fix which breaks existing behaviors/APIs)
- [ ] 🏗️ Code refactor
- [ ] ⚙️ Build configuration change
- [x] 📝 Documentation
- [x] 🧹 Chore / Housekeeping

## Small demo of new future-annotation functionality in example app

Simulator Screen Recording - iPhone 14 Pro - 2023-01-23 at 12 23 03

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,6 +5,7 @@ publish: ## publish all projects to pub.dev
 
 example_run_build_runner: ## Generate code for the example app inside the widget_driver folder.
 	bash scripts/run-example-run-build-runner.sh
+	make format
```
Comment (robert-northmind) [inline Makefile:8]: It makes sense to format the newly generated code according to our formatting rules.

File: widget_driver/README.md [hunk 2]
```diff
@@ -271,7 +281,7 @@ This will automatically make sure that your widget get reloaded and it can consu
 Here is a demo of how the final product looks like
 
 <p>
-  <img src="doc/resources/counter_widget_demo.gif?raw=true"
+  <img src="https://github.com/bmw-tech/widget_driver/blob/master/widget_driver/doc/resources/counter_widget_demo.gif?raw=true"
```
Comment (robert-northmind) [inline widget_driver/README.md:284]: This fixes the bug on pub.dev where the image resource was not found

File: widget_driver/example/lib/widgets/coffee_consumption/coffee_counter/coffee_counter_widget_driver.dart [hunk 3]
```diff
@@ -33,15 +34,37 @@ class CoffeeCounterWidgetDriver extends WidgetDriver {
...
 
   @TestDriverDefaultValue()
-  void consumeCoffee() {
+  void consumeCoffeeQuick() {
+    _consumptionService.consumedOneCoffee();
+  }
+
+  /// Returns true if we could consume a coffee slow.
+  /// We cannot consume slow if we are already consuming a slow coffee.
+  @TestDriverDefaultFutureValue(false)
```
Comment (robert-northmind) [inline widget_driver/example/lib/widgets/coffee_consumption/coffee_counter/coffee_counter_widget_driver.dart:54]: Show cases how to use the new future annotations and also how to combine annotations together with documentation

File: widget_driver/README.md [hunk 4]
```diff
@@ -217,6 +223,7 @@ class CounterWidget extends DrivableWidget<CounterWidgetDriver> {
 And voila, we are done!  
 Now check out that widget code! Isn't that clean.  
 Just focused on UI. No StreamBuilders or BuildContext being watched.  
+No business logic in you view code!
```
Comment (4ndrey) [inline widget_driver/README.md:226]:
```suggestion
No business logic in your view code!
```

File: widget_driver/README.md [hunk 5]
```diff
@@ -171,7 +177,7 @@ Now we can continue to the next step!
 ### 2: Code generation
 
 Now it is time to generate some code so that we get our `TestDriver` and `WidgetDriverProvider` set up for us.  
-*If you prefer to do this the old fashioned way without code generation, then that is also possible. 
+*If you prefer to do this the old fashioned way without code generation, then that is also possible.
```
Comment (4ndrey) [inline widget_driver/README.md:180]:
```suggestion
*If you prefer to do this the old fashioned way without code generation, then that is also possible.
```

File: widget_driver/doc/widget_drivers.md [hunk 6]
```diff
@@ -120,11 +120,11 @@ You get two option here for how to resolve these dependencies. Either you can gr
 **Now what about those annotations?**  
 They are needed for the `testDrivers` to be generated correctly. So for all properties and methods which you expose to the widget, you will need to add these annotations.
 
-For properties you add `@DriverProperty({default_value})`.  
+For properties and methods you add `@TestDriverDefaultValue({default_value})`.  
 The `default_value` should be the default value which you want to use when this widget is being created by other widgets under test.
 
-For methods you add `@DriverAction()`.  
-This is needed so that any potential parent widget when tested can call these methods without crashing.
+For properties and methods which returns futures, then you add `@TestDriverDefaultFutureValue({default_value})`.
```
Comment (4ndrey) [inline widget_driver/doc/widget_drivers.md:126]:
```suggestion
For properties and methods which returns futures, you have to add `@TestDriverDefaultFutureValue({default_value})`.  
```

**Review by 4ndrey — COMMENTED** [review]
Looks good, added some minor corrections

File: widget_driver/README.md (hunk 5 above)
Comment (robert-northmind) [reply widget_driver/README.md:180]: Adding 2 spaces just makes a bit nicer formatting.. 2 spaces forces a new line. 
If you remove the 2 spaces and do a new line in code, then that new line is ignored by the markdown readers..

**Review by 4ndrey — APPROVED** [review]

**Review by JanMichaelPeter — APPROVED** [review]

**Pull request merged**